internal/node/node_repositories.bzl

```diff
@@ -176,44 +178,57 @@ exports_files([
 alias(name = "node", actual = "{node}")
 alias(name = "npm", actual = "{npm}")
 alias(name = "yarn", actual = "{yarn}")
+filegroup(
+  name = "node_runfiles",
+  srcs = glob(
+    [
+      "bin/node.js",
+      "{node_dir}/**",
+      "{yarn_dir}/**",
+    ],
+    exclude = [
+      "**/*.md",
+      "**/*.html",
+    ],
+  ),
+)
 """.format(
     node = node_entry,
     npm = npm_node_repositories_entry,
-    yarn = yarn_node_repositories_entry))
+    yarn = yarn_node_repositories_entry,
+    node_dir = NODE_DIR,
+    yarn_dir = YARN_DIR))
 
   # The entry points for node for osx/linux and windows
   if not is_windows:
     # Sets process.env['PATH'] for node, npm & yarn and runs user script
     # This extra step is needed as process.env['PATH'] needs to be set
     # in some cases on osx/linux and in other cases PATH set in
-    # bin/node is sufficient.
+    # bin/node is sufficient. The first argument is the PATH to prepend.
     repository_ctx.file("bin/node.js", content="""//Generated by node_repositories.bzl
-const {{spawn}} = require('child_process');
-process.env['PATH'] = `"{root}":${{process.env['PATH']}}`;
-const proc = spawn("{node}", process.argv.slice(2), {{stdio: [process.stdin, process.stdout, process.stderr]}});
-proc.on('close', (code) => {{ process.exit(code); }});
-""".format(
-    root = repository_ctx.path("bin"),
-    node = node_exec))
+const {spawn} = require('child_process');
+process.env['PATH'] = `${process.argv[2]}:${process.env['PATH']}`;
+const proc = spawn(process.argv0, process.argv.slice(3), {stdio: [process.stdin, process.stdout, process.stderr]});
+proc.on('close', (code) => { process.exit(code); });
+""")
 
     # Sets PATH and runs bin/node.js passing all arguments
-    repository_ctx.file("bin/node", content="""
-export PATH="{root}":$PATH
-"{node}" "{script}" "$@"
+    repository_ctx.file("bin/node", content="""#!/bin/bash
```

Comment:
can you please change the shell to `/usr/bin/env bash`? Please see #247 for context.

Reply:
That PR is hung up on Windows compatibility issues, I think we should rebase later.
Also that one has no tests so we won't know if we broke it :(